File: DS/LinkedList/1/GetValOfNode.py
# Find the node at the given position counting backwards from the tail

class Node:
    def __init__(self,data,next):
        self.data= data
        self.next = next

class LinkedList:
    def __init__(self):
        self.currnode = None
    
    def add(self,data):
        newNode = Node(data,self.currnode)
        self.currnode = newNode
        return self.currnode
    
    def printlist(self,head):
        while head:
            print head.data
            head = head.next
    
    def getVal(self,head,pos):
        tempnode = head
        currpos = 0
        while head:
            if currpos > pos:
                tempnode = tempnode.next
            head = head.next
            currpos += 1
        
        print tempnode.data
    
if __name__ == "__main__":
    li = LinkedList()
    li.add(2)
    li.add(1)
    node1 = li.add(3)
    li.printlist(node1)
    print "getVal"
    li.getVal(node1,2)

File: DS/LinkedList/GetValOfNode.js
// Find the node at the given position counting backwards from the tail

function Node(data, next) {
    this.data = data;
    this.next = next;
}

function LinkedList() {
    this.currnode = null;
}

LinkedList.prototype = {
    add: function add(data) {
        var newNode = new Node(data, this.currnode);
        this.currnode = newNode;
        return this.currnode;
    },
    print: function(head) {
        while (head) {
            console.log(head.data);
            head = head.next;
        }
    },
    getVal: function(head, pos) {
        var tempNode = head;
        var currpos = 0;
        while (head) {
            if (currpos > pos) {
                tempNode = tempNode.next;
            }
            head = head.next;
            currpos++;
        }
        console.log(tempNode.data);
    }
}

var li = new LinkedList();
li.add(2);
li.add(1);
var node1 = li.add(3);
li.print(node1);
li.getVal(node1, 0);